jujugui/static/gui/src/test/test_bakery.js

```diff
@@ -66,6 +66,36 @@ describe('Bakery', function() {
     assert.equal(bakery.visitMethod, newVisitMethod);
   });
 
+  it('can accept an already baked macaroon', function() {
+    bakery = new Y.juju.environments.web.Bakery({
+      webhandler: new Y.juju.environments.web.WebHandler(),
+      serviceName: 'test',
+      macaroon: 'foo-bar'
+    });
+    assert.equal(bakery.getMacaroon(), 'foo-bar');
+  });
+
+  it('can be configured to use localStorage', function() {
+    bakery = new Y.juju.environments.web.Bakery({
+      webhandler: new Y.juju.environments.web.WebHandler(),
+      serviceName: 'test',
+      macaroon: 'foo-bar',
+      nonCookie: true
+    });
+    assert.equal(window.localStorage.getItem('Macaroons-test'), 'foo-bar');
```

Comment:
This is leaky for test suites, see my comment above about passing in localStorage